R/helper_functions.R

```diff
@@ -195,6 +195,36 @@ check_bestpreds_preds <- function(modelLibrary){
 #####################################################
 # Extraction functions
 #####################################################
+#' @title Extract the method name associated with a single train object
+#' @description Extracts the method name associated with a single train object.  Note
+#' that for standard models (i.e. those already prespecified by caret), the
+#' "method" attribute on the train object is used directly while for custom
+#' models the "method" attribute within the model$modelInfo attribute is
+#' used instead.
+#' @param x a single caret train object
+#' @return Name associated with model
+extractModelName <- function(x) {
+  if (is.list(x$method)){
+    validateCustomModel(x$method)$method
+  } else if (x$method == "custom"){
+    validateCustomModel(x$modelInfo)$method
+  } else x$method
+}
+
+#' @title Validate a custom caret model info list
+#' @description Currently, this only ensures that all model info lists
+#' were also assigned a "method" attribute for consistency with usage
+#' of non-custom models
+#' @param x a model info list (e.g. \code{getModelInfo("rf", regex=F)\[[1]]})
```

Comment:
This looks correct to me, but doesn't match the `.Rd` file (see below).  I think running `devtools::document()` will fix the problem.

You can check your work locally with `devtools::check()`, which conveniently also runs `devtools::document()`.